lib/commands/access.js

```diff
@@ -1,223 +1,225 @@
-    const pkgName = await this.getPackage(pkg, false)
-    const collabs = await libaccess.lsCollaborators(pkgName, usr, opts)
+  async #getStatus (pkg) {
+    const pkgName = await this.#getPackage(pkg, false)
+    const visibility = await libnpmaccess.getVisibility(pkgName, this.npm.flatOptions)
+    this.#output({ [pkgName]: visibility.public ? 'public' : 'private' })
+  }
 
-    // TODO - print these out nicely (breaking change)
-    this.npm.output(JSON.stringify(collabs, null, 2))
+  async #set (subcmd, pkg) {
+    const [subkey, subval] = subcmd.split('=')
+    switch (subkey) {
+      case 'mfa':
+      case '2fa':
+        return this.#setMfa(pkg, subval)
+      case 'status':
+        return this.#setStatus(pkg, subval)
+    }
   }
 
-  async edit () {
-    throw new Error('edit subcommand is not implemented')
+  async #setMfa (pkg, level) {
+    const pkgName = await this.#getPackage(pkg, false)
+    await otplease(this.npm, this.npm.flatOptions, (opts) => {
+      return libnpmaccess.setMfa(pkgName, level, opts)
+    })
+    // TODO output?
   }
 
-  modifyPackage (pkg, opts, fn, requireScope = true) {
-    return this.getPackage(pkg, requireScope)
-      .then(pkgName => otplease(this.npm, opts, opts => fn(pkgName, opts)))
+  async #setStatus (pkg, status) {
+    // only scoped packages can have their access changed
+    const pkgName = await this.#getPackage(pkg, true)
+    if (status === 'private') {
+      status = 'restricted'
+    }
+    await otplease(this.npm, this.npm.flatOptions, (opts) => {
+      return libnpmaccess.setAccess(pkgName, status, opts)
+    })
+    return this.#getStatus(pkgName)
   }
 
-  async getPackage (name, requireScope) {
-    if (name && name.trim()) {
-      return name.trim()
-    } else {
+  async #getPackage (name, requireScope) {
+    if (!name) {
       try {
         const pkg = await readPackageJson(path.resolve(this.npm.prefix, 'package.json'))
         name = pkg.name
       } catch (err) {
         if (err.code === 'ENOENT') {
-          throw new Error(
-            'no package name passed to command and no package.json found'
-          )
+          throw Object.assign(new Error('no package name given and no package.json found'), {
+            code: 'ENOENT',
+          })
         } else {
           throw err
         }
       }
+    }
 
-      if (requireScope && !name.match(/^@[^/]+\/.*$/)) {
-        throw this.usageError('This command is only available for scoped packages.')
-      } else {
-        return name
+    const spec = npa(name)
+    if (requireScope && !spec.scope) {
+      throw this.usageError('This command is only available for scoped packages.')
+    }
+    return name
+  }
+
+  #output (items, limiter) {
+    const output = {}
+    for (const item in items) {
+      let val = items[item]
+      if (val === 'read') {
+        val = 'read-only'
+      } else if (val === 'write') {
+        val = 'read-write'
+      }
+      output[item] = val
+    }
```

Comment:
```suggestion
    const lookup = {
      read: 'read-only',
      write: 'read-write',
    }
    const output = Object.fromEntries(Object.entries(items).map(([item, val]) => [item, lookup[val] || val]))
```

Reply:
i find this change to be more difficult to read and very much prefer the original code here

Reply:
alternative:
```suggestion
    const lookup = {
      __proto__: null,
      read: 'read-only',
      write: 'read-write',
    }
    const output = {}
    for (const item in items) {
      const val = items[item]
      output[item] = lookup[val] || val
    }
```